src/components/flyout/deviceDetailFlyout.js

```diff
@@ -176,7 +184,13 @@ class DeviceDetailFlyout extends Component {
       this.setState({
         showAlarmsGrid: data.Items.length > 0,
         alarmRowData: data.Items.map(item => {
+          this.props.rulesAndActions.forEach(Rule => {
```

Comment:
As mentioned in the PR for the alarm list, you should preprocess the rulesAndActions when new props come in. It should create a map of `id` to `name` to avoid looping the rulesAndActions for every alarm.